lib/spack/spack/version.py

```diff
@@ -1284,7 +1284,7 @@ def repository_uri(self):
             return os.path.join(*components)
         except ValueError:
             # If it's not a git url, it's a local path
-            return os.path.abspath(self.pkg.git)
+            return llnl.util.filesystem.remove_file_mount(os.path.abspath(self.pkg.git))
```

Comment:
`repository_uri` should return a valid uri, and IIRC file uri have to be absolute paths. Additionally are you intentionally dropping the `://` component of the file url in that method?

Reply:
Repository_uri never returned a valid uri. It had a bit of a confusing name and should likely be renamed. It would return all of the pieces of the repository uri concatenated together to form a path.